secret-storage.js

```diff
@@ -0,0 +1,90 @@
+const nosql = require('nosql');
+const uuid = require('uuid/v4');
+
+const Storage = require('./storage.js');
+const NoSuchEntity = Storage.NoSuchEntity;
+exports.NoSuchEntity = NoSuchEntity;
+
+const crypto = require('./crypto.js');
+exports.DecryptionError = crypto.DecryptionError;
+
+const secretIndex = nosql.load('./secretIndex.db.json');
+
+exports.put = (name, secret, encryptionKey) => {
```

Comment:
I have decided not to worry too much about the data layer here. This was just some scaffolding for me to prototype off of. This will be replaced by implementations for things like Dynamo DB, Document DB, or Mongo. Do you think there are some concurrency issues I need to solve now?